### PR TITLE
005-tilegrid: calculate IOI base address

### PR DESCRIPTION
**Pull request by tmichalak**
Signed-off-by: Tomasz Michalak <tmichalak@antmicro.com>

#746 showed that we have some unknown bits for zynq. Some analysis indicated that many of them belong to the IOI tiles.
The purpose of this PR is to calculate the baseaddresses for the IOI tiles.
It's WIP because the solution either contains constants or too many candidates.

### CODE REVIEW

**Comment by litghost** [comment]
> Signed-off-by: Tomasz Michalak [tmichalak@antmicro.com](mailto:tmichalak@antmicro.com)
> 
> #746 showed that we have some unknown bits for zynq. Some analysis indicated that many of them belong to the IOI tiles.
> The purpose of this PR is to calculate the baseaddresses for the IOI tiles.
> It's WIP because the solution either contains constants or too many candidates.

It is my belief that the IOI and IOB offsets and base addresses are identical, so the initial IOI fuzzers have been assigning bits to the IOB segbits.

**Comment by tmichalak** [comment]
I guess the IOI base address calculation is ready for review. As @litghost predicted the base addresses and offsets are identical for IOBs and IOIs.
Now it's time for the ILOGIC and OLOGIC fuzzers.

File: fuzzers/005-tilegrid/generate_full.py [hunk 1]
```diff
@@ -317,6 +317,65 @@ def propagate_IOB_SING(database, tiles_by_grid):
...
+    Unlike most tiles, they do not behave consistently.  The tile at the top
+    of the column is the bottom half of a full IOB, and the tile at the bottom
+    of the column is the top half of a full IOB. The tile contains half of the sites
+    that are present in the full IOI, namely one ILOGIC, OLOGIC and IDELAY.
+    For this reason, explicit bit aliasing is used to map the full IOI bits into the two halves, and a
+    mapping is provided for the site naming.
+
+    """
+
+    seen_iois = set()
```
Comment (litghost) [inline fuzzers/005-tilegrid/generate_full.py]: You just copy pasted this code from propagate_IOB_SING, try again?

File: fuzzers/005-tilegrid/ioi/top.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        .DATAOUT(idelay_{site})
+        );
+    assign {owire} = {onet};
+    assign {net} = idelay_{site};
+       '''.format(
+            onet=luts.get_next_output_net(),
+            net=luts.get_next_input_net(),
+            **p),
+        file=connects)
+    if p['isone']:
```
Comment (litghost) [inline fuzzers/005-tilegrid/ioi/top.py]: This fuzzer doesn't find a base address for IOI at all? It literally toggles something in the IOB portion, not the IOI.  Try again?

**Comment by tmichalak** [comment]
This PR showed that the IOI base address calculation will need a total rework, closing for now.

**Pull request closed without merging**